control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1103,56 +1103,84 @@ func (r *HostedControlPlaneReconciler) reconcileKubeAPIServer(ctx context.Contex
-		return p.ReconcileExternalKubeconfigSecret(externalKubeconfigSecret, clientCertSecret, rootCA)
+		return kas.ReconcileExternalKubeconfigSecret(externalKubeconfigSecret, clientCertSecret, rootCA, p.OwnerRef, p.ExternalURL(), p.ExternalKubeconfigKey())
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile external kubeconfig secret: %w", err)
 	}
 
 	bootstrapKubeconfigSecret := manifests.KASBootstrapKubeconfigSecret(hcp.Namespace)
 	if _, err := controllerutil.CreateOrUpdate(ctx, r, bootstrapKubeconfigSecret, func() error {
-		return p.ReconcileBootstrapKubeconfigSecret(bootstrapKubeconfigSecret, bootstrapClientCertSecret, rootCA)
+		return kas.ReconcileBootstrapKubeconfigSecret(bootstrapKubeconfigSecret, bootstrapClientCertSecret, rootCA, p.OwnerRef, p.ExternalURL())
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile bootstrap kubeconfig secret: %w", err)
 	}
 
 	kubeAPIServerAuditConfig := manifests.KASAuditConfig(hcp.Namespace)
 	if _, err := controllerutil.CreateOrUpdate(ctx, r, kubeAPIServerAuditConfig, func() error {
-		return p.ReconcileAuditConfig(kubeAPIServerAuditConfig)
+		return kas.ReconcileAuditConfig(kubeAPIServerAuditConfig, p.OwnerRef, p.AuditPolicyProfile())
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile api server audit config: %w", err)
 	}
 
 	kubeAPIServerConfig := manifests.KASConfig(hcp.Namespace)
 	if _, err := controllerutil.CreateOrUpdate(ctx, r, kubeAPIServerConfig, func() error {
-		return p.ReconcileConfig(kubeAPIServerConfig)
+		return kas.ReconcileConfig(kubeAPIServerConfig,
+			p.OwnerRef,
+			p.ExternalIPConfig(),
+			p.ClusterNetwork(),
+			p.ServiceNetwork(),
+			p.NamedCertificates(),
+			p.APIServerPort,
+			p.TLSSecurityProfile(),
+			p.AdditionalCORSAllowedOrigins(),
+			p.InternalRegistryHostName(),
+			p.ExternalRegistryHostNames(),
+			p.DefaultNodeSelector(),
+			p.AdvertiseAddress,
+			p.ServiceAccountIssuerURL(),
+			p.CloudProvider,
+			p.CloudProviderConfig,
+			p.EtcdURL,
+			p.FeatureGates(),
+			p.ServiceNodePortRange(),
+		)
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile api server config: %w", err)
 	}
 
 	oauthMetadata := manifests.KASOAuthMetadata(hcp.Namespace)
 	if _, err := controllerutil.CreateOrUpdate(ctx, r, oauthMetadata, func() error {
-		return p.ReconcileOauthMetadata(oauthMetadata)
+		return kas.ReconcileOauthMetadata(oauthMetadata, p.OwnerRef, p.ExternalOAuthAddress, p.ExternalOAuthPort)
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile oauth metadata: %w", err)
 	}
 
 	kubeAPIServerDeployment := manifests.KASDeployment(hcp.Namespace)
 	if _, err := controllerutil.CreateOrUpdate(ctx, r, kubeAPIServerDeployment, func() error {
-		return p.ReconcileKubeAPIServerDeployment(kubeAPIServerDeployment)
+		return kas.ReconcileKubeAPIServerDeployment(kubeAPIServerDeployment,
+			p.OwnerRef,
+			p.DeploymentConfig,
+			p.NamedCertificates(),
+			p.CloudProviderConfig,
+			releaseImage.ComponentImages()["cluster-config-operator"],
```

Comment:
For this one, I think a struct like `KASImages` with fields for every image makes more sense and avoids errors in ordering of the parameters.